extensions/replication/ReplicationMetric.js

```diff
@@ -0,0 +1,80 @@
+const { Logger } = require('werelogs');
+
+const MetricsModel = require('../../lib/models/MetricsModel');
+
+class ReplicationMetric {
+    constructor() {
+        this._log = new Logger('ReplicationMetric');
+    }
+
+    withProducer(producer) {
+        this._producer = producer;
+        return this;
+    }
+
+    withSite(site) {
+        this._site = site;
+        return this;
+    }
+
+    withObjectSize(objectSize) {
+        this._objectSize = objectSize;
+        return this;
+    }
+
+    /**
+     * Setter for an instance of ActionQueueEntry.
+     * @param {ActionQueueEntry} entry - The entry to publish a metric for
+     * @return {ReplicationMetric} - The current instance
+     */
+    withEntry(entry) {
+        this._entry = entry;
+        return this;
+    }
+
+    withMetricType(type) {
+        this._metricsType = type;
+        return this;
+    }
+
+    withExtension(extension) {
+        this._extension = extension;
+        return this;
+    }
+
+    _isLifecycleAction() {
+        const { origin } = this._entry.getContext();
+        return origin !== undefined && origin === 'lifecycle';
```

Comment:
nice!

Reply:
I'm not a fan of this to be honest :wink: but for now it's fine for me, eventually I would prefer that the replication metrics can "dispatch" based on some explicit originator value, but without having any semantic about a particular originator in the metrics code itself, as it should be agnostic of who is using the replication service.